src/language-js/parser-babylon.js

```diff
@@ -111,6 +111,20 @@ function tryCombinations(fn, combinations) {
   throw error;
 }
 
+function rethrowSomeRecoveredErrors(ast) {
+  for (const error of ast.errors) {
+    if (
+      typeof error.message === "string" &&
+      error.message.startsWith(
+        "Adjacent JSX elements must be wrapped in an enclosing tag."
```

Comment:
@nicolo-ribaudo Would be more convenient if errors had some kind of short codes in order for such checks to not rely on the wording of the message.